src/js/common-event-bindings.js

```diff
@@ -0,0 +1,162 @@
+     * components `selectors` option block.
+     *
+     * @typedef {Object} MarkupEventBindings
+     */
+
+    fluid.registerNamespace("fluid.decoratorViewComponent");
+
+    //
+    // The methods below might be generic enough to go straight to infusion
+    //
+
+    /**
+     *
+     * Expands string encoded arguments to the event invoker, filling
+     * in any compact string versions of infusion invokers.
+     *
+     * @param {Object} that - The component itself.
+     * @param {String|Object} arg - A single argument being passed to the event
+     * infoker.
+     * @param {String} name - The name for the invoker.
+     * @return {Object} The expanded argument.
+     */
+    fluid.expandCompoundArg = function (that, arg, name) {
+        var expanded = arg;
+        if (typeof(arg) === "string") {
+            if (arg.indexOf("(") !== -1) {
+                var invokerec = fluid.compactStringToRec(arg, "invoker");
+                // TODO: perhaps a a courtesy we could expose {node} or even {this}
+                expanded = fluid.makeInvoker(that, invokerec, name);
+            } else {
+                expanded = fluid.expandOptions(arg, that);
+            }
+        }
+        return expanded;
+    };
+
+    /**
+     *
+     * Processes a single markup event binding decorator of event type jQuery.
+     * Currently the only type of event supported is jQuery events, so this does
+     * all the work.
+     *
+     * @param {MarkupEventBinding} dec - The single binding decorator being processed.
+     * @param {DOMNode} node - The node we are listening to for the specified events.
+     * @param {Object} that - The component itself.
+     * @param {String} name - Name that will be used for the invoker created to handle
+     * this event.
+     * @return {Array} List of event functions created for this binding decorator.
+     */
+    fluid.processjQueryDecorator = function (dec, node, that, name) {
+        var args = fluid.makeArray(dec.args);
+        var expanded = fluid.transform(args, function (arg, index) {
+            return fluid.expandCompoundArg(that, arg, name + " argument " + index);
+        });
+        fluid.log("Got expanded value of ", expanded, " for jQuery decorator");
+        // Support for listing multiple methods in an array, or just a single string method
+        var methods = [dec.method];
+        var togo = [];
+        if (fluid.isArrayable(dec.method)) {
+            methods = dec.method;
+        }
+        fluid.each(methods, function (method) {
+            var func = node[method];
+            togo.push(func.apply(node, expanded));
+        });
+        return togo;
+    };
+
+    /**
+     *
+     * Function to process the markup binding decorators and create the events described
+     * by them. The markup needs to be rendered and settled before this can be called.
+     *
+     * @param {Object} that - The component itself.
```

Comment:
I am the guiltiest party here, I do this all the time, but in later work, @amb26 and I have been adding the grade to these for purposes of clarity.

Reply:
Make these more specific using the component grade for the type.